src/components/MoneyRequestAmountInput.tsx

```diff
@@ -135,6 +139,7 @@ function MoneyRequestAmountInput(
     });
 
     const forwardDeletePressedRef = useRef(false);
+    const resetFlag = useRef(false);
```

Comment:
Let's call this the same as the new prop (resetClicked/shouldResetAmount/or whichever it ends up being called)